ek9000App/src/ekDiag.cpp

```diff
@@ -384,5 +384,6 @@ size_t COE_DecodeDiagString(void* string, char* outbuf, unsigned int outbuflen)
 			break;
 	}
 	strncpy(outbuf, tmpbuf, outbuflen);
+	outbuf[outbuflen-1] = 0;
```

Comment:
Pesky `strncpy`...

Reply:
I gotta add something like [this](https://github.com/ValveSoftware/source-sdk-2013/blob/master/mp/src/public/tier0/protected_things.h#L48) to all of my C++ projects lol

Reply:
Their string library is one of the few tolerable things in the Source engine codebase :)

I just grepped through epics base looking for `strncpy`, it could probably use some fixes. *Most* instances of `strncpy` seem to be followed by code to properly null terminate the string, but I also see some potential oob writes... ie `pdest[len] = 0` :thinking:

Reply:
Oh boy... I'd bet they'd welcome those fixes, or even an issue pointing out suspicious ones.

Reply:
Ended up opening a PR on epics base to fix the few potentially buggy strncpy calls in there